packages/completer/src/manager.ts

```diff
@@ -151,6 +151,19 @@ export class CompletionProviderManager implements ICompletionProviderManager {
     }
   }
 
+  /**
+   * Dismiss the completer in the widget with provided id.
+   *
+   * @param id - the id of notebook panel, console panel or code editor.
+   */
+  dismiss(id: string): void {
+    const handler = this._panelHandlers.get(id);
+    if (handler) {
+      // handler.completer.dispose();
+      handler.dispose();
```

Comment:
All this could probably be avoided if the documentsearch:end command could be inhibited in this context... case 27 should handle all ESC cases on the completer anyway. I'll do better!